src/mcnp/mcnp6/CMakeLists.txt

```diff
@@ -68,34 +89,64 @@ set(CMAKE_Fortran_FLAGS_RELWITHDEBINFO "-O1 -g")
 
 # C compiler flags
 if (CMAKE_C_COMPILER_ID STREQUAL "Intel")
-  #set(CMAKE_C_FLAGS "${CMAKE_Fortran_FLAGS} -mcmodel=medium")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pc64")
+  if (MCNP_VERSION STREQUAL "620")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -m64")
+  else ()
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pc64")
+  endif ()
 else ()
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -m64")
+  if (MCNP_VERSION STREQUAL "620")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-common")
+  endif ()
+endif ()
+
+# CXX compiler flags
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
+  if (MCNP_VERSION STREQUAL "620")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m64")
+  else ()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pc64")
+  endif ()
+else ()
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m64")
+  if (MCNP_VERSION STREQUAL "620")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-common")
+  endif ()
 endif ()
 
 # Fortran compiler flags
 if (CMAKE_Fortran_COMPILER_ID STREQUAL "Intel")
-  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -no-vec")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -heap-arrays 1024")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -traceback")
-  #set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -mcmodel=medium")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -pc64")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -diag-disable 8290,8291,8577")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -r8")
+  if (MCNP_VERSION STREQUAL "620")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -free")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -standard-semantics")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -m64")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -error-limit 1000")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -warn nodeclarations")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -stand f03")
+  else ()
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -no-vec")
+  endif ()
 else ()
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -m64")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -mieee-fp")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdollar-ok")
-  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fno-sign-zero")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdefault-real-8")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdefault-double-8")
-endif ()
-
-# Include directories
-include_directories(${CMAKE_CURRENT_LIST_DIR}/Source/import)
-if (MCNP_VERSION STREQUAL "611")
-  include_directories(${CMAKE_CURRENT_LIST_DIR}/Source/import/cgm)
+  if (MCNP_VERSION STREQUAL "620")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -ffree-form")
+    # OpenMPI is not Fortran 2003 compliant
+    if (NOT BUILD_MCNP_MPI)
+      set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -std=f2003")
```

Comment:
Commenting out this line makes DAG-MCNP6.2 compilable with gcc/g++/gfortran 5 and 6

Reply:
This fixed the build error I was getting :+1: